src/unix/os390.c

```diff
@@ -20,6 +20,498 @@
  */
 
 #include "internal.h"
+#include <sys/ioctl.h>
+#include <net/if.h>
+#include <utmpx.h>
+#include <unistd.h>
+#include <sys/ps.h>
+#if defined(__clang__)
+#include "csrsic.h"
+#else
+#include "//'SYS1.SAMPLIB(CSRSIC)'"
+#endif
+
+#define CVT_PTR           0x10
+#define CSD_OFFSET        0x294
+
+/* 
+    Long-term average CPU service used by this logical partition,
+    in millions of service units per hour. If this value is above
+    the partition's defined capacity, the partition will be capped.
+    It is calculated using the physical CPU adjustment factor
+    (RCTPCPUA) so it may not match other measures of service which
+    are based on the logical CPU adjustment factor. It is available
+    if the hardware supports LPAR cluster.
+*/
+#define RCTLACS_OFFSET    0xC4
+
+/* 32-bit count of alive CPUs. This includes both CPs and IFAs */
+#define CSD_NUMBER_ONLINE_CPUS        0xD4
+
+/* ADDRESS OF SYSTEM RESOURCES MANAGER (SRM) CONTROL TABLE */
+#define CVTOPCTP_OFFSET   0x25C
+
+/* Address of the RCT table */
+#define RMCTRCT_OFFSET    0xE4
+
+/* "V(IARMRRCE)" - ADDRESS OF THE RSM CONTROL AND ENUMERATION AREA. */
+#define CVTRCEP_OFFSET    0x490
+
+/* 
+    NUMBER OF FRAMES CURRENTLY AVAILABLE TO SYSTEM. 
+    EXCLUDED ARE FRAMES BACKING PERM STORAGE, FRAMES OFFLINE, AND BAD FRAMES
+*/
+#define RCEPOOL_OFFSET    0x004
+
+/* TOTAL NUMBER OF FRAMES CURRENTLY ON ALL AVAILABLE FRAME QUEUES. */
+#define RCEAFC_OFFSET     0x088
+
+typedef unsigned data_area_ptr_assign_type;
+
+typedef union {
+  struct {
+#if defined(_LP64)
+    data_area_ptr_assign_type lower;
+#endif
+    data_area_ptr_assign_type assign;
+  };
+  char* deref;
+} data_area_ptr; 
+
+void uv_loadavg(double avg[3]) {
+
+  /* TODO: implement the following */
+  avg[0] = 0;
+  avg[1] = 0;
+  avg[2] = 0;
+}
+
+int uv__platform_loop_init(uv_loop_t* loop) {
+  int fd;
+
+  fd = epoll_create1(UV__EPOLL_CLOEXEC);
+
+  if (fd != -1)
+    uv__cloexec(fd, 1);
+
+  loop->backend_fd = fd;
+
+  if (fd == -1)
+    return -errno;
+
+  return 0;
+}
+
+void uv__platform_loop_delete(uv_loop_t* loop) {
+  loop->backend_fd = -1;
+}
+
+uint64_t uv__hrtime(uv_clocktype_t type) {
+  uint64_t G = 1000000000;
```

Comment:
use these constants directly in the return statement? maybe also use scientific notation for conciseness.